Swc 135 implementar notificaciones por web socket para conexi n y desconexi n en lobbies

MatiasScantamburlo opened the pull request:
BONUS: ¡Integración realizada!

MatiasScantamburlo committed:
Se agrega ws /ws/lobby/{game_id}

- Este ws se encarga de notificar a los usuarios conectados dentro de un juego
    cuando otro usuario se conecta o desconecta.
- Hace uso de ManagerTypes.JOIN_LEAVE como manejador de ws

 En la rama SWC-135-Implementar-notificaciones-por-WebSocket-para-conexi-n-y-desconexi-n-en-lobbies
 Cambios a ser confirmados:
	modificados:     main.py
	nuevos archivos: tests/test_notify_lobby.py

MatiasScantamburlo committed:
Format

KaiserOfNone commented on main.py:
```diff
@@ -506,3 +506,40 @@ async def turn_change_notifier(websocket: WebSocket, game_id: int):
+            players_raw = game.players
+            players = [{"id": p.id, "name": p.name} for p in players_raw]
+
+            await manager.broadcast({"players": players}, game_id)
```
Ojo, en el endpoint de salir y entrar estoy mandando esto por ese ws
```python
    await join_leave_manager.broadcast(
        {
            "player_id": player_exit.id,
            "action": "join",
            "player_name": player_exit.name,
        },
        game.id,
    )
```
quizas debas acomodarlo para que tengan la forma de este, no es mala idea que manden la lista actualizada.

seedLou replied:
Desde el endpoint join_game tambien se llama al broadcast :0

KaiserOfNone commented on tests/test_notify_lobby.py:
```diff
@@ -0,0 +1,94 @@
+                }
+
+                # Se une el Lou^2
+                self.game_1.players.append(self.players[1])
```
usa `add_player`, no apendes a la lista de prepo. hay data secudaria que se genera y cuando lo haces asi no se crea.

KaiserOfNone reviewed (commented):
All good, solo un par de comments chiquitos

LourdesMamani00 commented on main.py:
```diff
@@ -506,3 +506,40 @@ async def turn_change_notifier(websocket: WebSocket, game_id: int):
+    Se espera: {user_identifier: 'valor'}
+    """
+    manager = Managers.get_manager(ManagerTypes.JOIN_LEAVE)
+    await manager.connect(websocket, game_id)
```
no se si aca va a cambiar con la nueva implementacion

LourdesMamani00 commented on main.py:
```diff
@@ -506,3 +506,40 @@ async def turn_change_notifier(websocket: WebSocket, game_id: int):
+            await manager.broadcast({"players": players}, game_id)
+
+    except WebSocketDisconnect:
+        manager.disconnect(websocket, game_id)
```
aca posiblemente cambie con la nueva implementacion, no se'

MatiasScantamburlo committed:
Merge branch 'develop' into SWC-135-Implementar-notificaciones-por-WebSocket-para-conexi-n-y-desconexi-n-en-lobbies

MatiasScantamburlo committed:
Se arregla FAILURE del merge con develop

    - connect y disconnect de connection_manager ahora necesitan un
      player_id
    - Se solicita ahora en /ws/lobby/{game_id} que se envie el player_id
      como variable. En PR discutir si es la mejor forma
    - Se corrige el test con dichos cambios

 En la rama SWC-135-Implementar-notificaciones-por-WebSocket-para-conexi-n-y-desconexi-n-en-lobbies

 Cambios a ser confirmados:
	modificados:     main.py
	modificados:     tests/test_notify_lobby.py

MatiasScantamburlo committed:
Se realizan los cambios sugeridos por Andy en el PR

    - Usamos nomeclatura "player_id" y "name_id"
    - Sustituimos append() por add_player()

 En la rama SWC-135-Implementar-notificaciones-por-WebSocket-para-conexi-n-y-desconexi-n-en-lobbies

 Cambios a ser confirmados:
	modificados:     main.py
	modificados:     tests/test_notify_lobby.py
